platform/darwin/src/MGLShapeSource.mm

```diff
@@ -84,6 +86,66 @@
         geoJSONOptions.cluster = value.boolValue;
     }
 
+    if (NSDictionary *value = options[MGLShapeSourceOptionClusterProperties]) {
+        if (![value isKindOfClass:[NSDictionary<NSString *, NSArray *> class]]) {
+            [NSException raise:NSInvalidArgumentException
+                        format:@"MGLShapeSourceOptionClusterProperties must be an NSDictionary with an NSString as a key and an array containing two NSExpression objects as a value."];
+        }
+
+        NSEnumerator *stringEnumerator = [value keyEnumerator];
+        NSString *key;
+
+        while (key = [stringEnumerator nextObject]) {
+            NSArray *expressionsArray = value[key];
+            if (![expressionsArray isKindOfClass:[NSArray class]]) {
+                [NSException raise:NSInvalidArgumentException
+                            format:@"MGLShapeSourceOptionClusterProperties dictinary member value must be an array containing two objects."];
+            }
+            // check that array has only 2 values
```

Comment:
Could you please change the comment to provide more context on why it should be 2 values